version: "3.8"

services:
  ui-tests:
    image: tp:v1
    build:
      context: .
      dockerfile: Dockerfile
    shm_size: 2gb
    volumes:
      - ./:/usr/workspace
      - /dev/shm:/dev/shm
    working_dir: /usr/workspace
    command: >
      /bin/sh -c "\
        pytest -sv -o log_cli=true --alluredir=allure-results -n=2 || true"
    tty: true
    environment:
      LOGIN: ${LOGIN}
      PASSWORD: ${PASSWORD}
      MANAGER_LOGIN: ${MANAGER_LOGIN}
      MANAGER_PASSWORD: ${MANAGER_PASSWORD}

  report:
    image: tp:v1
    build:
      context: .
      dockerfile: Dockerfile
    volumes:
      - ./:/usr/workspace
    working_dir: /usr/workspace
    command: >
      /bin/sh -c "\
        ls -la && \
        ls -la allure-results || echo 'No allure-results directory found!' && \
        allure generate allure-results --clean -o allure-report"